app/domain/authentication/resource_restrictions/extract_resource_restrictions.rb

```diff
@@ -7,6 +7,7 @@ module ResourceRestrictions
       dependencies: {
         resource_restrictions_class: Authentication::ResourceRestrictions::ResourceRestrictions,
         get_restriction_from_annotation: Authentication::ResourceRestrictions::GetRestrictionFromAnnotation,
+        fetch_resource_annotations: Authentication::ResourceRestrictions::FetchResourceAnnotations,
```

Comment:
the `.new` should be here

Reply:
Done

Reply:
Done

Reply:
well this should be fixed everywhere. when a command class is a dependency then we put the `.new` in the dependency and not in the code itself so that it is initialized only once.